server/src/main/java/com/cloud/deploy/DeploymentPlanningManagerImpl.java

```diff
@@ -1411,7 +1411,7 @@ private boolean canAvoidCluster(Cluster clusterVO, ExcludeList avoids, ExcludeLi
 
             if (vmRequiresSharedStorage) {
                 // check shared pools
-                List<StoragePoolVO> allPoolsInCluster = _storagePoolDao.findPoolsByTags(clusterVO.getDataCenterId(), clusterVO.getPodId(), clusterVO.getId(), null, false, 0);
+                List<StoragePoolVO> allPoolsInCluster = _storagePoolDao.findPoolsByTags(clusterVO.getDataCenterId(), clusterVO.getPodId(), clusterVO.getId(), ScopeType.CLUSTER, null, false, 0);
```

Comment:
@harikrishna-patnala can you check if this will skip the zone-wide pools of the cluster, or these are considered elsewhere?

Reply:
I've not changed the logic of the existing method regarding the scope, but I just generalized the method so that anyone can call this method with any scope. Previously also the internal method findPoolsByTags() used hardcoded ScopeType.CLUSTER in it.